### PR TITLE
Msg storage contract

### PR DESCRIPTION
**Pull request by RCantu92**
Test directory for BerfChatStorage.sol contract file.

### CODE REVIEW

File: contracts/BerfChatStorage.sol [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+    /// @notice Takes the message sent and stores it by
+    /// linking it to the chat id, comprised of the two
+    /// participant's addresses.
+    /// @param _to Address of the message recipient.
+    function sendMessage(address _to) public {
+        address _from = msg.sender;
+
+        if(existingChat[_from][_to] == true) {
```
Comment (fabdarice) [inline contracts/BerfChatStorage.sol:40]: let's test this.
```
if (_from > _to) {
    chatId = hashAddresses(_from, _to);
} else {
    chatId = hashAddresses(_to, _from);
}
```

File: contracts/BerfChatStorage.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        // DATA LOCATION VERIFICATION
+        return _addressHash;
+    }
+
+
+    /// @notice Takes the message sent and stores it by
+    /// linking it to the chat id, comprised of the two
+    /// participant's addresses.
+    /// @param _to Address of the message recipient.
+    function sendMessage(address _to) public {
```
Comment (fabdarice) [inline contracts/BerfChatStorage.sol:37]: missing the message

File: contracts/BerfChatStorage.sol (hunk 1 above)
Comment (fabdarice) [reply contracts/BerfChatStorage.sol:40]: If this works, you don't need the storage `existingChat`

File: test/BerfChatStorage.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    });
+    */
+
+    it("compares the generated chatId from sendMessage() from different msg.sender's", async function() {
+        // Get BerfChatStorage to deploy
+        const berfChatStorageFactory = await ethers.getContractFactory("BerfChatStorage");
+
+        // Pull array of Signers and
+        // assign to 'accounts'
+        const accounts = await ethers.getSigners();
```
Comment (fabdarice) [inline test/BerfChatStorage.ts:98]: `const [deployer, bob, alice, bad, ...accounts] = await ethers.getSigners()`

Comment (fabdarice) [reply test/BerfChatStorage.ts:98]: You should use the `before` statement from hardhat.

File: test/BerfChatStorage.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import { Signer } from "ethers";
+import { BerfChatStorage } from "../typechain/BerfChatStorage";
+
+chai.use(solidity);
+
+const { expect } = chai;
+
+describe("BerfChatStorage contract tests", function () {
+    let berfChatStorage: BerfChatStorage;
+
```
Comment (fabdarice) [inline test/BerfChatStorage.ts:13]: let firstAccount, secondAccount;

**Comment by RCantu92** [comment]
Updated to reflect comments made by Fabrice

**Pull request merged**